### PR TITLE
1592 committer not shown in versions

### LINKED ISSUE #1592: Committer not shown in versions
see http://ontohub.org/sandbox/engine2///ontology_versions. I have made all commits, but my name is not shown.

### PR DESCRIPTION
**Pull request by eugenk**
This shall fix #1592. It improves and makes more use of the `Commit` model.

### CODE REVIEW

File: db/migrate/20160614065105_add_association_from_commit_to_user.rb [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      attrs = select_attributes(ontology_version, :user_id, :commit_oid, :name)
+      commit = Commit.where(commit_oid: attrs[:commit_oid]).first
+      update_columns(commit,
+                     pusher_id: attrs[:user_id], pusher_name: attrs[:name])
+    end
+
+    up_split_name_and_email
+  end
+
+  def down_data
```
Comment (eugenk) [inline db/migrate/20160614065105_add_association_from_commit_to_user.rb]: Fixed with ff74840.

File: db/migrate/20160614065105_add_association_from_commit_to_user.rb [hunk 2]
```diff
@@ -0,0 +1,100 @@
+class AddAssociationFromCommitToUser < MigrationWithData
+  def up
```
Comment (eugenk) [inline db/migrate/20160614065105_add_association_from_commit_to_user.rb]: No, thanks.

File: db/migrate/20160614065105_add_association_from_commit_to_user.rb [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    add_index :commits, :author_id
+    add_index :commits, :committer_id
+    add_index :commits, :pusher_id
+
+    up_data
+
+    remove_column :ontology_versions, :user_id
+  end
+
+  def down
```
Comment (eugenk) [inline db/migrate/20160614065105_add_association_from_commit_to_user.rb]: No, thanks.

File: app/models/user.rb [hunk 4]
```diff
@@ -100,4 +109,27 @@ def check_remaining_admins
     end
   end
 
+  def remove_associations_from_commits
```
Comment (eugenk) [inline app/models/user.rb]: No, thanks. This one is very simple.

**Comment by eugenk** [comment]
I messed up the data on develop.ontohub.org, because I deployed a broken migration on this machine. The migration is fixed now. The broken data is still on develop.onothub.org, though.

Please test the migrations with `rake db:migrate:with_data` on your machines, because they can destroy data (`OntologyVersion#user_id` is removed). Keep in mind, that not all OntologyVersions are associated with a commit on the deployment machines.

File: db/data/20160614065104_create_missing_commits.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
+class CreateMissingCommits < MigrationWithData
+  def self.up
```
Comment (eugenk) [inline db/data/20160614065104_create_missing_commits.rb]: Fixed with 86856a6

File: db/migrate/20160614065105_add_association_from_commit_to_user.rb [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    add_column :ontology_versions, :user_id, :integer
+
+    down_data
+
+    remove_index :commits, :author_id
+    remove_index :commits, :committer_id
+    remove_index :commits, :pusher_id
+
+    # We emphasize "null: true" because there may be no user
```
Comment (phyrog) [inline db/migrate/20160614065105_add_association_from_commit_to_user.rb]: Copy & Paste error?

**Comment by phyrog** [comment]
Looks good to me fo now. Please rebase on staging, since there seem to be conflicts.

**Comment by phyrog** [comment]
Currently re-running the travis-ci/push; it failed because of Travis, so if it works now :+1:

**Comment by phyrog** [comment]
CI works, so :+1:

**Pull request merged**